src/app/GitUI/Editor/FileViewer.cs

```diff
@@ -1599,6 +1583,35 @@ private void resetSelectedLinesToolStripMenuItem_Click(object sender, EventArgs
             ResetSelectedLines();
         }
 
+        /// <summary>
+        /// Update the current blob id for the GitItemStatus.
+        /// TreeId is immutable for normal commits, must always be updated before use for Index.
+        /// TreeId is irrelevant for worktree (if dirty), but this sets IsSubmodule.
+        /// TODO: add to GitModule, similar to GetFileBlobHash
+        /// </summary>
+        public ObjectId? TryUpdateTreeId(GitItemStatus file,
```

Comment:
The .NET idiomatic use of `TryXxx` is to return boolean indicating whether the operation succeeded and an "out" value parameter.

```cs
        public bool TryUpdateTreeId(GitItemStatus file,
            ObjectId? objectId,
            CancellationToken cancellationToken,
            [NotNullIf(true)] out ObjectId? <name>)
```

Reply:
Xmldocs for the args would be great to have as well.